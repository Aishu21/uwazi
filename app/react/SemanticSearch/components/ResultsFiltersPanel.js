import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Form } from 'react-redux-form';
import { NumericRangeSlide } from 'app/ReactReduxForms';


import SidePanel from 'app/Layout/SidePanel';
import { t } from 'app/I18N';


export function ResultsFiltersPanel({ open, filtersValues }) {
  const filters = [
    { label: 'Threshold', model: 'threshold', min: 0.05, max: 1, step: 0.05 },
    { label: 'Minimum relevant sentences', model: 'minRelevantSentences', min: 1, max: 50, step: 1 },
    { label: 'Minimum relevant score', model: 'minRelevantScores', min: 5, max: 100, step: 5 }
  ];

  return (
    <SidePanel open={open}>
      <div className="sidepanel-body">
        <div className="sidepanel-title">
          { t('System', 'Fine tune')}
        </div>
        <Form model="semanticSearch.resultsFilters">
          <div className="view">
            { filters.map(filter => (
              <dl className="metadata-type-text" key={filter.label}>
                <dt>{t('System', filter.label)} {filtersValues[filter.model]}</dt>
                <dd>
                  <NumericRangeSlide model={`.${filter.model}`} prefix={filter.model} min={filter.min} max={filter.max} step={filter.step} />
                </dd>
              </dl>
            ))}
          </div>
        </Form>
      </div>
    </SidePanel>
  );
}

ResultsFiltersPanel.propTypes = {
  open: PropTypes.bool.isRequired,
  filtersValues: PropTypes.object.isRequired,
};

function mapStateToProps({ semanticSearch }) {
  return {
    open: semanticSearch.selectedDocument.isEmpty(),
    filtersValues: semanticSearch.resultsFilters
  };
}

export default connect(mapStateToProps)(ResultsFiltersPanel);
