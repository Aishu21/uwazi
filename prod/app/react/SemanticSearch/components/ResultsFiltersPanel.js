"use strict";Object.defineProperty(exports, "__esModule", { value: true });exports.ResultsFiltersPanel = ResultsFiltersPanel;exports.default = void 0;var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _reactRedux = require("react-redux");
var _reactReduxForm = require("react-redux-form");
var _ReactReduxForms = require("../../ReactReduxForms");
var _UI = require("../../UI");

var _SidePanel = _interopRequireDefault(require("../../Layout/SidePanel"));
var _I18N = require("../../I18N");function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}var REACT_ELEMENT_TYPE;function _jsx(type, props, key, children) {if (!REACT_ELEMENT_TYPE) {REACT_ELEMENT_TYPE = typeof Symbol === "function" && Symbol["for"] && Symbol["for"]("react.element") || 0xeac7;}var defaultProps = type && type.defaultProps;var childrenLength = arguments.length - 3;if (!props && childrenLength !== 0) {props = { children: void 0 };}if (props && defaultProps) {for (var propName in defaultProps) {if (props[propName] === void 0) {props[propName] = defaultProps[propName];}}} else if (!props) {props = defaultProps || {};}if (childrenLength === 1) {props.children = children;} else if (childrenLength > 1) {var childArray = new Array(childrenLength);for (var i = 0; i < childrenLength; i++) {childArray[i] = arguments[i + 3];}props.children = childArray;}return { $$typeof: REACT_ELEMENT_TYPE, type: type, key: key === undefined ? null : '' + key, ref: null, props: props, _owner: null };}

const filters = [
{
  label: 'Threshold',
  model: 'threshold',
  min: 0.3,
  max: 1,
  step: 0.01,
  minLabel: 'Exploration',
  maxLabel: 'Precision' },

{
  label: 'Minimum relevant sentences per document',
  model: 'minRelevantSentences',
  min: 1,
  max: 50,
  step: 1,
  minLabel: '',
  maxLabel: '' }];



const filterValue = (filter, filtersValues) => filter.model === 'threshold' ?
`${(filtersValues[filter.model] * 100).toFixed(2)}%` : filtersValues[filter.model];

function ResultsFiltersPanel({ open, filtersValues }) {
  return (
    _jsx(_SidePanel.default, { open: open }, void 0,
    _jsx("div", { className: "sidepanel-body" }, void 0,
    _jsx("div", { className: "sidepanel-title" }, void 0,
    (0, _I18N.t)('System', 'Fine tune')),

    _jsx(_reactReduxForm.Form, { model: "semanticSearch.resultsFilters" }, void 0,
    _jsx("div", { className: "view" }, void 0,
    filters.map((filter) =>
    _jsx("dl", { className: "metadata-type-text" }, filter.label,
    _jsx("dt", {}, void 0, (0, _I18N.t)('System', filter.label), " ", filterValue(filter, filtersValues)),
    _jsx("dd", {}, void 0,
    _jsx(_ReactReduxForms.NumericRangeSlide, {
      delay: 200,
      model: `.${filter.model}`,
      prefix: filter.model,
      min: filter.min,
      max: filter.max,
      step: filter.step,
      minLabel: (0, _I18N.t)('System', filter.minLabel),
      maxLabel: (0, _I18N.t)('System', filter.maxLabel) })))))),






    _jsx("div", { className: "semantic-search-help" }, void 0,
    _jsx("div", { className: "alert alert-info alert-vertical" }, void 0,
    _jsx(_UI.Icon, { icon: "info-circle", size: "2x" }),
    _jsx("p", {}, void 0,
    _jsx(_I18N.Translate, { translationKey: "Semantic search overview" }, void 0, "Semantic search is a technique to provide contextual results. Its ability to capture concepts and word associations in human language enables the retrieval of related information such as synonyms, connected categories or entities, etc. .")),





    _jsx("p", {}, void 0,
    _jsx(_I18N.Translate, { translationKey: "Semantic search threshold help" }, void 0, "The threshold determines how close the results match the search concept. Move the slider to the right to narrow down the concept of the search query. The obtained results will be more precise. Move the slider to the left to more broaden the concept and explore related content.")),






    _jsx("p", {}, void 0,
    _jsx(_I18N.Translate, { translationKey: "Semantic search minimum sentences help" }, void 0, "Semantic search is applied to each sentence in a document. Filter the documents by the minimum number of sentences that exceed the threshold.")))))));









}








function mapStateToProps({ semanticSearch }) {
  return {
    open: semanticSearch.selectedDocument.isEmpty(),
    filtersValues: semanticSearch.resultsFilters };

}var _default =

(0, _reactRedux.connect)(mapStateToProps)(ResultsFiltersPanel);exports.default = _default;