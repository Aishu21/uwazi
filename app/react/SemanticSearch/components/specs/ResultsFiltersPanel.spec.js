import React from 'react';
import { shallow } from 'enzyme';

import { ResultsFiltersPanel } from '../ResultsFiltersPanel';

describe('ResultsFiltersPanel', () => {
  let props;
  let component;
  beforeEach(() => {
    props = {
      open: true,
      filtersValues: { threshold: 10 }
    };

    component = shallow(<ResultsFiltersPanel {...props}/>);
  });

  describe('render', () => {
    it('should render a result', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
