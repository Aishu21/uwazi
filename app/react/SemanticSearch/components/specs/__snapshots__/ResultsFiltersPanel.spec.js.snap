// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFiltersPanel render should render a result 1`] = `
<SidePanel
  open={true}
>
  <div
    className="sidepanel-body"
  >
    <div
      className="sidepanel-title"
    >
      Fine tune
    </div>
    <Connect(Form)
      model="semanticSearch.resultsFilters"
    >
      <div
        className="view"
      >
        <dl
          className="metadata-type-text"
          key="Threshold"
        >
          <dt>
            Threshold
             
            1000%
          </dt>
          <dd>
            <NumericRangeSlide
              delay={200}
              max={1}
              min={0.3}
              model=".threshold"
              prefix="threshold"
              step={0.01}
            />
          </dd>
        </dl>
        <dl
          className="metadata-type-text"
          key="Minimum relevant sentences"
        >
          <dt>
            Minimum relevant sentences
             
          </dt>
          <dd>
            <NumericRangeSlide
              delay={200}
              max={50}
              min={1}
              model=".minRelevantSentences"
              prefix="minRelevantSentences"
              step={1}
            />
          </dd>
        </dl>
      </div>
    </Connect(Form)>
  </div>
</SidePanel>
`;
